index.js

```diff
@@ -1,11 +1,26 @@
 'use strict';
 const fs = require('fs');
 const path = require('path');
+const stream = require('stream');
 const uniqueString = require('unique-string');
 const tempDir = require('temp-dir');
+const isStream = require('is-stream');
+const {writeFile} = fs.promises;
 
 const getPath = () => path.join(tempDir, uniqueString());
 
+const writeStream = async (filePath, data) => new Promise((resolve, reject) => {
+	const writable = fs.createWriteStream(filePath);
+
+	stream.pipeline(data, writable, error => {
+		if (error) {
+			reject(error);
+			data.unpipe(writable);
+			writable.end();
```

Comment:
Are you sure it's necessary to do this unpipe and .end() manually? I would have assumed the `pipeline` does it.

Reply:
Fixed!